examples/distributed_stencil/1d_stencil.cpp

```diff
@@ -0,0 +1,169 @@
+    U[1] = data_type(Nx, 0.0, alloc);
+
+    init(U, Nx, rank, num_localities);
+
+    // setup communicator
+    using communicator_type = communicator<double>;
+    communicator_type comm(rank, num_localities);
+
+    if (rank == 0)
+    {
+        std::cout << "Starting benchmark with " << num_localities << 
+                     " nodes and " << num_worker_threads <<
+                     " threads.\n";
+    }
+
+    if (comm.has_neighbor(communicator_type::left))
+    {
+        // Send initial value to the left neighbor
+        comm.set(communicator_type::left, U[0][0], 0);
+    }
+    if (comm.has_neighbor(communicator_type::right))
+    {
+        // Send initial value to the right neighbor
+        comm.set(communicator_type::right, U[0][Nx-1], 0);
+    }
+    
+    auto range = boost::irange(static_cast<std::size_t>(0), nlp);
+
+    executor_type executor(numa_domains);
+    auto policy = hpx::parallel::execution::par.on(executor);
+
+    hpx::util::high_resolution_timer t;
+    for (std::size_t t = 0; t < steps; ++t)
+    {
+        data_type& curr = U[t % 2];
+        data_type& next = U[(t + 1) % 2];
+
+        hpx::parallel::for_each(
+            policy,
+            std::begin(range), std::end(range),
+            [&U, local_nx, nlp, t] (std::size_t i)
+            {
+                if (i == 0)     stencil_update(U, 1, local_nx, t);
+                else if (i == nlp-1)    stencil_update(U, i * local_nx, (i + 1) * local_nx - 1, t);
+                else    stencil_update(U, i * local_nx, (i + 1) * local_nx, t);
+            }
+        );
+
+        // Left and right neighbor element should be transported
+        // by now.
+
+        if (comm.has_neighbor(communicator_type::left))
+        {
+            // Synchronization step. Wait for the left neighbor to arrive.
+            double left = comm.get(communicator_type::left, t).get();
+
+            // next[0] = (left + curr[1]) * 0.5f;
+            next[0] = curr[0] + ((k*dt)/(dx*dx)) * (left - 2*curr[0] + curr[1]);
+
+            // Dispatch the updated value to left neighbor for it to get
+            // consumed in the next timestep
+            comm.set(communicator_type::left, next[0], t+1); 
+        }
+
+        if (comm.has_neighbor(communicator_type::right))
+        {
+            // Synchronization step. Wait for the right neighbor to arrive.
+            double right = comm.get(communicator_type::right, t).get();
+
+            // next[Nx-1] = (right + curr[Nx-2]) * 0.5f;
+            next[Nx-1] = curr[Nx-1] + ((k*dt)/(dx*dx)) * (curr[Nx-2] - 2*curr[Nx-1] + right);
+
+            // Dispatch the updated value to right neighbor for it to get
+            // consumed in the next timestep
+            comm.set(communicator_type::right, next[Nx-1], t+1); 
+        }
+    }
+    double elapsed = t.elapsed();
+    double telapsed = t_main.elapsed();
+
+    // for (double elem : U[(steps)%2])
+    //    std::cout << elem << " ";
+    // std::cout << std::endl;
```

Comment:
Remove? Or if you'd like a verbose mode that could be behind a command line flag.